### PR TITLE
Added gelfand tsetlin polytope

### PR DESCRIPTION
**Pull request by tbrysiewicz**
I only added the one function 
```gelfand_tsetlin```
For some reason, ```Polymake.polytope.gelfand_tsetlin([3,2,1])``` complains:
```
ERROR: Error processing type expression Polytope: Integer is not a suitable coordinate type at /home/taylor/.julia/artifacts/337994e1146a8a47c31b601eaa87acf37794bcb9/share/polymake/apps/common/rules/basic_types line 420.
```

so I'm forced to do
 ```Polymake.polytope.gelfand_tsetlin(Array{Rational,1}([3,2,1]))```
 instead, unless there's an alternative suggestion.

### CODE REVIEW

File: src/PolyhedralGeometry/Polyhedron/standard_constructions.jl [hunk 1]
```diff
@@ -841,3 +841,31 @@ true
...
+A polyhedron in ambient dimension 3
+
+julia> isfulldimensional(p)
+true
+
+julia> volume(p)
+8
+```
+"""
+#It doesn't like the integer type so I have to convert it to Rational. Why?
```
Comment (lkastner) [inline src/PolyhedralGeometry/Polyhedron/standard_constructions.jl]: This commented line seems to confuse the documenter. If you move it below the function declaration, the error should go away.

**Review by lkastner — APPROVED** [review]

**Comment by lkastner** [comment]
> I only added the one function `gelfand_tsetlin` For some reason, `Polymake.polytope.gelfand_tsetlin([3,2,1])` complains:
> 
> ```
> ERROR: Error processing type expression Polytope: Integer is not a suitable coordinate type at /home/taylor/.julia/artifacts/337994e1146a8a47c31b601eaa87acf37794bcb9/share/polymake/apps/common/rules/basic_types line 420.
> ```
> 
> so I'm forced to do `Polymake.polytope.gelfand_tsetlin(Array{Rational,1}([3,2,1]))` instead, unless there's an alternative suggestion.

Yes, this construction is templated in polymake, so you can put other scalar types there. However Integer is not ok and I guess for now Rational will do. Probably @alexej-jordan will have a look at this with his upcoming PR.

**Comment by benlorenz** [comment]
> > I only added the one function `gelfand_tsetlin` For some reason, `Polymake.polytope.gelfand_tsetlin([3,2,1])` complains:
> > ```
> > ERROR: Error processing type expression Polytope: Integer is not a suitable coordinate type at /home/taylor/.julia/artifacts/337994e1146a8a47c31b601eaa87acf37794bcb9/share/polymake/apps/common/rules/basic_types line 420.
> > ```
> > so I'm forced to do `Polymake.polytope.gelfand_tsetlin(Array{Rational,1}([3,2,1]))` instead, unless there's an alternative suggestion.
> 
> Yes, this construction is templated in polymake, so you can put other scalar types there. However Integer is not ok and I guess for now Rational will do. Probably @alexej-jordan will have a look at this with his upcoming PR.

To be more precise: this is a bug in how the `type_upgrade` for this function is written in polymake and should be fixed there, it needs an additional `[ is_ordered_field(type_upgrade<Scalar, Rational>) ]` to make sure the type is a field since polytopes with integer coordinates are not possible.

Until that is fixed I would say enforcing rational here is probably be fine.
If we want to support additional types before the next polymake release we could just keep the type in the generic case, but add an extra function which forces an explicit conversion to Rational for any integer types.

File: src/PolyhedralGeometry/Polyhedron/standard_constructions.jl [hunk 2]
```diff
@@ -841,3 +841,31 @@ true
...
+gelfand_tsetlin(λ::AbstractVecOrMat) = Polyhedron(Polymake.polytope.gelfand_tsetlin(Array{Rational,1}(λ),projected=false))
```
Comment (benlorenz) [inline src/PolyhedralGeometry/Polyhedron/standard_constructions.jl]: A few remarks:
- `AbstractVecOrMat` is not really true, this polymake function will only accept a vector and no matrix types.
- I think non-ASCII characters are discouraged (forbidden?) in Oscar
- `Vector{T}` is preferred over `Array{T,1}`

Comment (thofma) [reply src/PolyhedralGeometry/Polyhedron/standard_constructions.jl]: Yes, no non-ASCII in the "source" (although it is allowed when displaying stuff, see https://github.com/oscar-system/Oscar.jl/issues/318).

**Comment by micjoswig** [comment]
For OSCAR it is perfectly OK to always assume that \lambda (and thus the resulting polytope) is rational.  So casting to `Rational` is fine.  There is a reason why this is more general in polymake (because of ongoing research), but none of this needs to be reflected in OSCAR (at least not in the foreseeable future).

File: src/PolyhedralGeometry/Polyhedron/standard_constructions.jl (hunk 2 above)
Comment (tbrysiewicz) [reply src/PolyhedralGeometry/Polyhedron/standard_constructions.jl]: Great. Thank you!

**Review by benlorenz — APPROVED** [review]

File: src/PolyhedralGeometry/Polyhedron/standard_constructions.jl [hunk 3]
```diff
@@ -841,3 +841,30 @@ true
...
 """
 project_full(P::Polyhedron) = Polyhedron(Polymake.polytope.project_full(pm_object(P)))
+
+
+
+@doc Markdown.doc"""
+
+    gelfand_tsetlin(lambda::AbstractVector)
+
+Construct the Gelfand Tsetlin polytope indexed by lambda where lambda is weakly decreasing vector.
```
Comment (fingolfin) [inline src/PolyhedralGeometry/Polyhedron/standard_constructions.jl]:
```suggestion
Construct the Gelfand Tsetlin polytope indexed by a weakly decreasing vector `lambda`.
```

**Pull request merged**